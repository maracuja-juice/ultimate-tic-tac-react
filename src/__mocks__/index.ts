import {
  unfinishedBoardMock,
  movesForUnfinishedBoardMock,
} from './unfinishedBoardMock';
import {
  crossFinishedGameMock,
  circleFinishedGameMock,
} from './finishedGameDataMock';
import {
  circleFinishedBoardMock,
  crossFinishedBoardMock,
  movesForCircleFinishedBoardMock,
  movesForCrossFinishedBoardMock,
} from './finishedBoardMock';
import {
  activeBoardsForBoardWithTwoMovesMock,
  boardWithTwoMovesMock,
} from './boardWithTwoMovesMock';
import {
  boardWithThreeMovesMock,
  activeBoardsForBoardWithThreeMovesMock,
  movesForBoardWithThreeMovesMock,
} from './boardWithThreeMovesMock';
import {
  boardWithOneMoveMock,
  activeBoardsForBoardWithOneMoveMock,
} from './boardWithOneMoveMock';

export {
  unfinishedBoardMock,
  movesForUnfinishedBoardMock,
  crossFinishedGameMock,
  circleFinishedGameMock,
  circleFinishedBoardMock,
  movesForCircleFinishedBoardMock,
  crossFinishedBoardMock,
  movesForCrossFinishedBoardMock,
  boardWithTwoMovesMock,
  activeBoardsForBoardWithTwoMovesMock,
  boardWithThreeMovesMock,
  movesForBoardWithThreeMovesMock,
  activeBoardsForBoardWithThreeMovesMock,
  boardWithOneMoveMock,
  activeBoardsForBoardWithOneMoveMock,
};
