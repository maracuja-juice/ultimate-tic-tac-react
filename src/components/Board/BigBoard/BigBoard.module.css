.bigBoard {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-template-rows: repeat(3, 1fr);
  grid-gap: 1em;
  justify-content: center;
}
