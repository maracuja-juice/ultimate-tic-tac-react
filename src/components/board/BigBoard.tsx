import * as React from 'react';
import SmallBoardContainer from './SmallBoardContainer';

interface BigBoardProps {
}

interface BigBoardState {
}

export class BigBoard extends React.Component<BigBoardProps, BigBoardState> {

    constructor( props: BigBoardProps ) {
        super( props );

        this.createSmallBoards = this.createSmallBoards.bind( this );
    }

    createSmallBoards() {
        let rows = [];
        for (let i = 0; i < 3; i++) {
            for (let j = 0; j < 3; j++) {
                rows.push( (
                    <SmallBoardContainer
                        key={i.toString() + j.toString()}
                        x={i}
                        y={j}
                    />) );
            }
        }
        return rows;
    }

    render() {
        const smallBoards = this.createSmallBoards();

        return (
            <div className="big-board">
                {smallBoards}
            </div>
        );
    }
}