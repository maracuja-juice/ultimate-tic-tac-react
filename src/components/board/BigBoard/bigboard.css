.big-board {
    display: grid;
    grid-template-columns: 12.5vh 12.5vh 12.5vh;
    grid-template-rows: 12.5vh 12.5vh 12.5vh;
    grid-gap: 2vh;
}

@media(min-width: 35em) {
    .big-board {
        grid-template-columns: 20vh 20vh 20vh;
        grid-template-rows: 20vh 20vh 20vh;
        grid-gap: 4vh;
    }
}