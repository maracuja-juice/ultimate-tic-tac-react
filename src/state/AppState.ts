import { Point } from '../util';

export interface AppState {
  currentGame: GameInformation;
  finishedGames: FinishedGameState[];
  analysisGame: AnalysisGame;
}

export interface GameInformation {
  board: SmallBoardInformation[];
  game: GameState;
  moves: MoveState[];
  activeBoards: Point[];
}

export interface FinishedGameState {
  id?: string;
  date: string;
  winner: Winner;
  gameState: SmallBoardInformation[];
  moves: MoveState[];
  saveState: string;
  errorMessage: string;
}

export interface AnalysisGame {
  id: string;
  board: SmallBoardInformation[];
  game: GameState;
  moves: MoveState[];
  activeBoards: Point[];
  currentMove: number;
}

export interface GameState {
  currentPlayer: Player;
  isFinished: boolean;
  winningPlayer: Winner;
}

export interface TileInformation {
  position: Point;
  value: TileValue;
}

export interface SmallBoardInformation extends TileInformation {
  tiles: SmallTileInformation[];
}

export interface SmallTileInformation extends TileInformation {
  boardPosition: Point;
}

export enum TileValue {
  Cross = 0,
  Circle = 1,
  Empty = 2,
  Destroyed = 3,
}

export enum Winner {
  Cross = 0,
  Circle = 1,
  None = 2,
  Draw = 3,
}

export enum Player {
  Cross = 0,
  Circle = 1,
}

export interface Move {
  boardPosition: Point;
  tilePosition: Point;
}

export interface MoveState extends Move {
  player: Player;
  moveNumber: number;
}

export interface GenericAction {
  type: string;
  // tslint:disable-next-line: no-any
  payload?: any;
}

export interface MarkSpecially {
  condition: boolean;
  position?: { boardPosition: Point; tilePosition: Point };
}
