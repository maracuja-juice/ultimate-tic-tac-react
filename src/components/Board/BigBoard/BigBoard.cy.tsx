import * as React from 'react';
import { mount } from '@cypress/react';
import { BigBoard } from './BigBoard';
import { Player } from '../../../state/AppState';
import {
  activeBoardsForBoardWithOneMoveMock,
  activeBoardsForBoardWithTwoMovesMock,
  boardWithOneMoveMock,
  boardWithTwoMovesMock,
  crossFinishedBoardMock,
  drawFinishedBoardMock,
} from '../../../mocks';

describe('BigBoard', () => {
  it('One Move', () => {
    mount(
      <BigBoard
        currentPlayer={Player.Circle}
        board={boardWithOneMoveMock}
        activeBoards={activeBoardsForBoardWithOneMoveMock}
      />,
    );

    cy.matchImageSnapshot();
  });

  it('Two Moves', () => {
    mount(
      <BigBoard
        currentPlayer={Player.Cross}
        board={boardWithTwoMovesMock}
        activeBoards={activeBoardsForBoardWithTwoMovesMock}
      />,
    );

    cy.matchImageSnapshot();
  });

  it('Finished Cross', () => {
    mount(
      <BigBoard
        currentPlayer={Player.Cross}
        board={crossFinishedBoardMock}
        activeBoards={[]}
      />,
    );
    cy.matchImageSnapshot();
  });

  it('Finished Draw', () => {
    mount(
      <BigBoard
        currentPlayer={Player.Circle}
        board={drawFinishedBoardMock}
        activeBoards={[]}
      />,
    );
    cy.matchImageSnapshot();
  });
});
