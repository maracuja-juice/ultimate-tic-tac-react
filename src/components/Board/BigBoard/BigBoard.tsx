import * as React from 'react';
import { SmallBoard } from '../SmallBoard/SmallBoard';
import { AppState, Player, SmallBoardInformation } from '../../../state/AppState';

import './bigboard.css';
import { arePointsEqual } from '../../../util';
import { Point } from '../../../util/Point';
import { connect } from 'react-redux';
import { playerMoved } from '../../../state/currentGame/game/gameAction';

interface BigBoardProps {
    currentPlayer: Player;
    board: SmallBoardInformation[];
    activeBoards: Point[];
    onPlayerMoved: ( boardX: number, boardY: number, tileX: number, tileY: number ) => void;
    analysisMode?: boolean;
}

export class BigBoard extends React.Component<BigBoardProps> {

    constructor( props: BigBoardProps ) {
        super( props );

        this.createSmallBoards = this.createSmallBoards.bind( this );
        this.isMoveOnBoardAllowed = this.isMoveOnBoardAllowed.bind( this );
    }

    isMoveOnBoardAllowed( x: number, y: number, activeBoards: Point[] ) {
        if (!activeBoards) {
            return false;
        }

        const theBoardPlayedOnIsActive = activeBoards.some( board => arePointsEqual( {x, y}, board ) );
        return theBoardPlayedOnIsActive;
    }

    createSmallBoards() {
        const {currentPlayer, board, activeBoards, onPlayerMoved, analysisMode} = this.props;
        const rows = [];

        for (let x = 0; x < 3; x++) {
            for (let y = 0; y < 3; y++) {

                const smallBoard = board.find( tile => arePointsEqual( tile.position, {x, y} ) );

                if (smallBoard) {
                    const isMoveAllowed = this.isMoveOnBoardAllowed( x, y, activeBoards );

                    let onTileClicked;
                    if (analysisMode) {
                        // tslint:disable-next-line
                        onTileClicked = ( tileX: number, tileY: number ) => {
                        };
                    } else {
                        onTileClicked = ( tileX: number, tileY: number ) => {
                            onPlayerMoved( x, y, tileX, tileY );
                        };
                    }

                    rows.push(
                        <SmallBoard
                            key={`x: ${x}/ Y: ${y}`}
                            x={x}
                            y={y}
                            isMoveAllowed={isMoveAllowed}
                            currentPlayer={currentPlayer}
                            tiles={smallBoard.tiles}
                            winningPlayer={smallBoard.value}
                            onTileClicked={onTileClicked}
                        />
                    );
                }
            }
        }

        return rows;
    }

    render() {
        return (
            <div className="big-board">
                {this.createSmallBoards()}
            </div>
        );
    }
}

const mapStateToProps = ( state: AppState ) => ({
    currentPlayer: state.game.currentPlayer,
    board: state.board,
    activeBoards: state.activeBoards,
});

// tslint:disable-next-line: no-any
const mapDispatchToProps = ( dispatch: any ) => ({
    onPlayerMoved: ( boardX: number, boardY: number, tileX: number, tileY: number ) =>
        dispatch( playerMoved( {x: boardX, y: boardY}, {x: tileX, y: tileY} ) )
});

export default connect( mapStateToProps, mapDispatchToProps )( BigBoard );
