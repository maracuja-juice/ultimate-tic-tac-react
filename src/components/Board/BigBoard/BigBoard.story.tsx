import * as React from 'react';
import { object, select, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import { BigBoard } from './BigBoard';
import { Player } from '../../../state/AppState';
import {
  circleFinishedBoardMock,
  crossFinishedBoardMock,
  unfinishedBoardMock,
} from '../../../__mocks__';
import { action } from '@storybook/addon-actions';
import './BigBoard.module.css';

const stories = storiesOf('BigBoard', module);
stories.addDecorator(withKnobs);

const boardValues = { Circle: Player.Circle, Cross: Player.Cross };

stories.add('BigBoard, not finished', () => (
  <BigBoard
    currentPlayer={select('currentPlayer', boardValues, Player.Circle)}
    board={unfinishedBoardMock}
    activeBoards={object('activeBoards', [{ x: 0, y: 2 }])}
    onPlayerMoved={action('onPlayerMoved')}
  />
));

stories.add('BigBoard, finished circle', () => (
  <BigBoard
    currentPlayer={select('currentPlayer', boardValues, Player.Cross)}
    board={circleFinishedBoardMock}
    activeBoards={[]}
    onPlayerMoved={action('onPlayerMoved')}
  />
));

stories.add('BigBoard, finished cross', () => (
  <BigBoard
    currentPlayer={select('currentPlayer', boardValues, Player.Circle)}
    board={crossFinishedBoardMock}
    activeBoards={[]}
    onPlayerMoved={action('onPlayerMoved')}
  />
));

stories.add('BigBoard, markTileSpecially', () => (
  <BigBoard
    currentPlayer={select('currentPlayer', boardValues, Player.Cross)}
    board={circleFinishedBoardMock}
    activeBoards={[]}
    onPlayerMoved={action('onPlayerMoved')}
    markTileSpecially={object('markTileSpecially', {
      condition: true,
      position: {
        boardPosition: { x: 2, y: 2 },
        tilePosition: { x: 0, y: 1 },
      },
    })}
  />
));
