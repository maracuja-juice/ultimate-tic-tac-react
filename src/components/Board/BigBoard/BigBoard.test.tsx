import * as React from 'react';
import { render, screen } from '../../../test-utils';
import userEvent from '@testing-library/user-event';
import { BigBoard } from './BigBoard';
import { Player } from '../../../state/AppState';
import { emptyBoardMock, unfinishedBoardMock } from '../../../__mocks__';

test('should render buttons for board', () => {
  const playerMoved = jest.fn(() => {});
  const activeBoards = [{ x: 0, y: 0 }];

  render(
    <BigBoard
      currentPlayer={Player.Cross}
      board={emptyBoardMock}
      activeBoards={activeBoards}
      onPlayerMoved={playerMoved}
    />,
  );

  const buttons = screen.getAllByRole('button');
  const removeExtensions = (str: string) => str.replace(/\..*/, '');
  const buttonContent = buttons.map((b: HTMLElement) =>
    b.textContent !== null ? removeExtensions(b.textContent) : '',
  );
  expect(buttonContent).toHaveLength(9 * 9);
  const expectedContent = Array.from({ length: 9 * 9 }).map((_) => '');
  expect(buttonContent).toEqual(expectedContent);
});

test('allows clicking on buttons without content', () => {
  const playerMoved = jest.fn(() => {});
  const activeBoards = [
    { x: 0, y: 0 },
    { x: 0, y: 1 },
    { x: 0, y: 2 },
    { x: 1, y: 0 },
    { x: 1, y: 1 },
    { x: 1, y: 2 },
    { x: 2, y: 0 },
    { x: 2, y: 1 },
    { x: 2, y: 2 },
  ];

  render(
    <BigBoard
      currentPlayer={Player.Cross}
      board={unfinishedBoardMock}
      activeBoards={activeBoards}
      onPlayerMoved={playerMoved}
    />,
  );

  const buttonsWithContent = screen.getAllByRole('button', { name: /./ });
  buttonsWithContent.forEach((button) => userEvent.click(button));
  expect(playerMoved).not.toHaveBeenCalled();

  const emptyButtons = screen.getAllByRole('button', { name: /^$/ });
  emptyButtons.forEach((button) => userEvent.click(button));
  expect(playerMoved).toHaveBeenCalledTimes(emptyButtons.length);
});
