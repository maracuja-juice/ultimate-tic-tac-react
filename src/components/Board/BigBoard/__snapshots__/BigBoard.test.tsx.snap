// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BigBoard should match snapshot 1`] = `
<div
  className="bigBoard"
>
  <SmallBoard
    currentPlayer={0}
    key="x: 0/ Y: 0"
    moveAllowed={true}
    onTileClicked={[Function]}
    tiles={
      Array [
        Object {
          "boardPosition": Object {
            "x": 0,
            "y": 0,
          },
          "position": Object {
            "x": 0,
            "y": 0,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 0,
            "y": 0,
          },
          "position": Object {
            "x": 0,
            "y": 1,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 0,
            "y": 0,
          },
          "position": Object {
            "x": 0,
            "y": 2,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 0,
            "y": 0,
          },
          "position": Object {
            "x": 1,
            "y": 0,
          },
          "value": 0,
        },
        Object {
          "boardPosition": Object {
            "x": 0,
            "y": 0,
          },
          "position": Object {
            "x": 1,
            "y": 1,
          },
          "value": 0,
        },
        Object {
          "boardPosition": Object {
            "x": 0,
            "y": 0,
          },
          "position": Object {
            "x": 1,
            "y": 2,
          },
          "value": 0,
        },
        Object {
          "boardPosition": Object {
            "x": 0,
            "y": 0,
          },
          "position": Object {
            "x": 2,
            "y": 0,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 0,
            "y": 0,
          },
          "position": Object {
            "x": 2,
            "y": 1,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 0,
            "y": 0,
          },
          "position": Object {
            "x": 2,
            "y": 2,
          },
          "value": 2,
        },
      ]
    }
    winningPlayer={0}
    x={0}
    y={0}
  />
  <SmallBoard
    currentPlayer={0}
    key="x: 0/ Y: 1"
    moveAllowed={false}
    onTileClicked={[Function]}
    tiles={
      Array [
        Object {
          "boardPosition": Object {
            "x": 0,
            "y": 1,
          },
          "position": Object {
            "x": 0,
            "y": 0,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 0,
            "y": 1,
          },
          "position": Object {
            "x": 0,
            "y": 1,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 0,
            "y": 1,
          },
          "position": Object {
            "x": 0,
            "y": 2,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 0,
            "y": 1,
          },
          "position": Object {
            "x": 1,
            "y": 0,
          },
          "value": 0,
        },
        Object {
          "boardPosition": Object {
            "x": 0,
            "y": 1,
          },
          "position": Object {
            "x": 1,
            "y": 1,
          },
          "value": 0,
        },
        Object {
          "boardPosition": Object {
            "x": 0,
            "y": 1,
          },
          "position": Object {
            "x": 1,
            "y": 2,
          },
          "value": 0,
        },
        Object {
          "boardPosition": Object {
            "x": 0,
            "y": 1,
          },
          "position": Object {
            "x": 2,
            "y": 0,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 0,
            "y": 1,
          },
          "position": Object {
            "x": 2,
            "y": 1,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 0,
            "y": 1,
          },
          "position": Object {
            "x": 2,
            "y": 2,
          },
          "value": 2,
        },
      ]
    }
    winningPlayer={0}
    x={0}
    y={1}
  />
  <SmallBoard
    currentPlayer={0}
    key="x: 0/ Y: 2"
    moveAllowed={false}
    onTileClicked={[Function]}
    tiles={
      Array [
        Object {
          "boardPosition": Object {
            "x": 0,
            "y": 2,
          },
          "position": Object {
            "x": 0,
            "y": 0,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 0,
            "y": 2,
          },
          "position": Object {
            "x": 0,
            "y": 1,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 0,
            "y": 2,
          },
          "position": Object {
            "x": 0,
            "y": 2,
          },
          "value": 1,
        },
        Object {
          "boardPosition": Object {
            "x": 0,
            "y": 2,
          },
          "position": Object {
            "x": 1,
            "y": 0,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 0,
            "y": 2,
          },
          "position": Object {
            "x": 1,
            "y": 1,
          },
          "value": 0,
        },
        Object {
          "boardPosition": Object {
            "x": 0,
            "y": 2,
          },
          "position": Object {
            "x": 1,
            "y": 2,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 0,
            "y": 2,
          },
          "position": Object {
            "x": 2,
            "y": 0,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 0,
            "y": 2,
          },
          "position": Object {
            "x": 2,
            "y": 1,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 0,
            "y": 2,
          },
          "position": Object {
            "x": 2,
            "y": 2,
          },
          "value": 2,
        },
      ]
    }
    winningPlayer={2}
    x={0}
    y={2}
  />
  <SmallBoard
    currentPlayer={0}
    key="x: 1/ Y: 0"
    moveAllowed={false}
    onTileClicked={[Function]}
    tiles={
      Array [
        Object {
          "boardPosition": Object {
            "x": 1,
            "y": 0,
          },
          "position": Object {
            "x": 0,
            "y": 0,
          },
          "value": 0,
        },
        Object {
          "boardPosition": Object {
            "x": 1,
            "y": 0,
          },
          "position": Object {
            "x": 0,
            "y": 1,
          },
          "value": 1,
        },
        Object {
          "boardPosition": Object {
            "x": 1,
            "y": 0,
          },
          "position": Object {
            "x": 0,
            "y": 2,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 1,
            "y": 0,
          },
          "position": Object {
            "x": 1,
            "y": 0,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 1,
            "y": 0,
          },
          "position": Object {
            "x": 1,
            "y": 1,
          },
          "value": 1,
        },
        Object {
          "boardPosition": Object {
            "x": 1,
            "y": 0,
          },
          "position": Object {
            "x": 1,
            "y": 2,
          },
          "value": 0,
        },
        Object {
          "boardPosition": Object {
            "x": 1,
            "y": 0,
          },
          "position": Object {
            "x": 2,
            "y": 0,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 1,
            "y": 0,
          },
          "position": Object {
            "x": 2,
            "y": 1,
          },
          "value": 0,
        },
        Object {
          "boardPosition": Object {
            "x": 1,
            "y": 0,
          },
          "position": Object {
            "x": 2,
            "y": 2,
          },
          "value": 2,
        },
      ]
    }
    winningPlayer={2}
    x={1}
    y={0}
  />
  <SmallBoard
    currentPlayer={0}
    key="x: 1/ Y: 1"
    moveAllowed={false}
    onTileClicked={[Function]}
    tiles={
      Array [
        Object {
          "boardPosition": Object {
            "x": 1,
            "y": 1,
          },
          "position": Object {
            "x": 0,
            "y": 0,
          },
          "value": 1,
        },
        Object {
          "boardPosition": Object {
            "x": 1,
            "y": 1,
          },
          "position": Object {
            "x": 0,
            "y": 1,
          },
          "value": 1,
        },
        Object {
          "boardPosition": Object {
            "x": 1,
            "y": 1,
          },
          "position": Object {
            "x": 0,
            "y": 2,
          },
          "value": 0,
        },
        Object {
          "boardPosition": Object {
            "x": 1,
            "y": 1,
          },
          "position": Object {
            "x": 1,
            "y": 0,
          },
          "value": 1,
        },
        Object {
          "boardPosition": Object {
            "x": 1,
            "y": 1,
          },
          "position": Object {
            "x": 1,
            "y": 1,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 1,
            "y": 1,
          },
          "position": Object {
            "x": 1,
            "y": 2,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 1,
            "y": 1,
          },
          "position": Object {
            "x": 2,
            "y": 0,
          },
          "value": 1,
        },
        Object {
          "boardPosition": Object {
            "x": 1,
            "y": 1,
          },
          "position": Object {
            "x": 2,
            "y": 1,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 1,
            "y": 1,
          },
          "position": Object {
            "x": 2,
            "y": 2,
          },
          "value": 2,
        },
      ]
    }
    winningPlayer={1}
    x={1}
    y={1}
  />
  <SmallBoard
    currentPlayer={0}
    key="x: 1/ Y: 2"
    moveAllowed={false}
    onTileClicked={[Function]}
    tiles={
      Array [
        Object {
          "boardPosition": Object {
            "x": 1,
            "y": 2,
          },
          "position": Object {
            "x": 0,
            "y": 0,
          },
          "value": 1,
        },
        Object {
          "boardPosition": Object {
            "x": 1,
            "y": 2,
          },
          "position": Object {
            "x": 0,
            "y": 1,
          },
          "value": 1,
        },
        Object {
          "boardPosition": Object {
            "x": 1,
            "y": 2,
          },
          "position": Object {
            "x": 0,
            "y": 2,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 1,
            "y": 2,
          },
          "position": Object {
            "x": 1,
            "y": 0,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 1,
            "y": 2,
          },
          "position": Object {
            "x": 1,
            "y": 1,
          },
          "value": 1,
        },
        Object {
          "boardPosition": Object {
            "x": 1,
            "y": 2,
          },
          "position": Object {
            "x": 1,
            "y": 2,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 1,
            "y": 2,
          },
          "position": Object {
            "x": 2,
            "y": 0,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 1,
            "y": 2,
          },
          "position": Object {
            "x": 2,
            "y": 1,
          },
          "value": 0,
        },
        Object {
          "boardPosition": Object {
            "x": 1,
            "y": 2,
          },
          "position": Object {
            "x": 2,
            "y": 2,
          },
          "value": 2,
        },
      ]
    }
    winningPlayer={2}
    x={1}
    y={2}
  />
  <SmallBoard
    currentPlayer={0}
    key="x: 2/ Y: 0"
    moveAllowed={false}
    onTileClicked={[Function]}
    tiles={
      Array [
        Object {
          "boardPosition": Object {
            "x": 2,
            "y": 0,
          },
          "position": Object {
            "x": 0,
            "y": 0,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 2,
            "y": 0,
          },
          "position": Object {
            "x": 0,
            "y": 1,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 2,
            "y": 0,
          },
          "position": Object {
            "x": 0,
            "y": 2,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 2,
            "y": 0,
          },
          "position": Object {
            "x": 1,
            "y": 0,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 2,
            "y": 0,
          },
          "position": Object {
            "x": 1,
            "y": 1,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 2,
            "y": 0,
          },
          "position": Object {
            "x": 1,
            "y": 2,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 2,
            "y": 0,
          },
          "position": Object {
            "x": 2,
            "y": 0,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 2,
            "y": 0,
          },
          "position": Object {
            "x": 2,
            "y": 1,
          },
          "value": 0,
        },
        Object {
          "boardPosition": Object {
            "x": 2,
            "y": 0,
          },
          "position": Object {
            "x": 2,
            "y": 2,
          },
          "value": 2,
        },
      ]
    }
    winningPlayer={2}
    x={2}
    y={0}
  />
  <SmallBoard
    currentPlayer={0}
    key="x: 2/ Y: 1"
    moveAllowed={false}
    onTileClicked={[Function]}
    tiles={
      Array [
        Object {
          "boardPosition": Object {
            "x": 2,
            "y": 1,
          },
          "position": Object {
            "x": 0,
            "y": 0,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 2,
            "y": 1,
          },
          "position": Object {
            "x": 0,
            "y": 1,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 2,
            "y": 1,
          },
          "position": Object {
            "x": 0,
            "y": 2,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 2,
            "y": 1,
          },
          "position": Object {
            "x": 1,
            "y": 0,
          },
          "value": 1,
        },
        Object {
          "boardPosition": Object {
            "x": 2,
            "y": 1,
          },
          "position": Object {
            "x": 1,
            "y": 1,
          },
          "value": 1,
        },
        Object {
          "boardPosition": Object {
            "x": 2,
            "y": 1,
          },
          "position": Object {
            "x": 1,
            "y": 2,
          },
          "value": 1,
        },
        Object {
          "boardPosition": Object {
            "x": 2,
            "y": 1,
          },
          "position": Object {
            "x": 2,
            "y": 0,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 2,
            "y": 1,
          },
          "position": Object {
            "x": 2,
            "y": 1,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 2,
            "y": 1,
          },
          "position": Object {
            "x": 2,
            "y": 2,
          },
          "value": 2,
        },
      ]
    }
    winningPlayer={1}
    x={2}
    y={1}
  />
  <SmallBoard
    currentPlayer={0}
    key="x: 2/ Y: 2"
    moveAllowed={false}
    onTileClicked={[Function]}
    tiles={
      Array [
        Object {
          "boardPosition": Object {
            "x": 2,
            "y": 2,
          },
          "position": Object {
            "x": 0,
            "y": 0,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 2,
            "y": 2,
          },
          "position": Object {
            "x": 0,
            "y": 1,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 2,
            "y": 2,
          },
          "position": Object {
            "x": 0,
            "y": 2,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 2,
            "y": 2,
          },
          "position": Object {
            "x": 1,
            "y": 0,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 2,
            "y": 2,
          },
          "position": Object {
            "x": 1,
            "y": 1,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 2,
            "y": 2,
          },
          "position": Object {
            "x": 1,
            "y": 2,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 2,
            "y": 2,
          },
          "position": Object {
            "x": 2,
            "y": 0,
          },
          "value": 2,
        },
        Object {
          "boardPosition": Object {
            "x": 2,
            "y": 2,
          },
          "position": Object {
            "x": 2,
            "y": 1,
          },
          "value": 0,
        },
        Object {
          "boardPosition": Object {
            "x": 2,
            "y": 2,
          },
          "position": Object {
            "x": 2,
            "y": 2,
          },
          "value": 2,
        },
      ]
    }
    winningPlayer={2}
    x={2}
    y={2}
  />
</div>
`;
